### DISCUSSION TITLE
Problem to create item and update O2M item in the aftermath from Flow

### DISCUSSION THREAD

**Discussion by Corentin7301** [directus/directus#17229] [Archive]
### Describe the Bug

Hello !
I would like, when creating a user, to create a record in a table and link the "shaper" field (which is an 02M relationship) to my user according to the user ID.
To do this, I created an action trigger when creating a user.
Then, I create data in my "custom board info" table and give it:
```json
{
       "shaper" : {
        "id" : " $trigger.key "
    }
}
```

When I do this in Insomnia or Postman:

```gql
mutation createNewForm($id:ID!) {
  create_custom_board_infos_item(
    data: {shaper: {id: $id}}
  ) {
    id
  }
}
```

The item is created and the shaper is linked by its ID.
When I do :

```json
{
       "shaper" : {
        "id" : " $trigger.key "
    }
}
```

In my Flow, I have this error:

```json
{
  "status": 400,
  "code": "INVALID_PAYLOAD"
  "extensions": {}
}
```

Yet the shaper ID is recovered correctly.

I don't understand, can someone please help me?

### To Reproduce

1. create a field with M2O relation connected to another collection named `custom_board_infos` in user collection.
2. create a field in `custom_board_infos` named `shaper` connected to user collection.
3. create a flow with trigger is Create an item in directus_users collection
4. add Create datas operation. In custom_board_infos collection.
5. Add:
```json
{
       "shaper" : {
        "id" : $trigger.key"
    }
}
```
in datas.

### Errors Shown

```
{
  "status": 400,
  "code": "INVALID_PAYLOAD",
  "extensions": {}
}
```

### What version of Directus are you using?

9.22.1

### What version of Node.js are you using?

18.12.1 in local

### What database are you using?

Postgres 14

### What browser are you using?

Chrome

### How are you deploying Directus?

Locally and on Coolify (Docker)

**Comment by Corentin7301** [comment]
Hello, I am still blocked 😢, I take the liberty of asking for help again! 🙏🏼
Thanks a lot! 🙏🏼🔥

**Comment by rijkvanzanten** [comment]
Have you tried

```json
{
  "shaper" : {
    "id" : "{{$trigger.key}}"
  }
}
```

? The `{{}}` indicate it's a variable instead of a string.

**Comment by Corentin7301** [comment]
@rijkvanzanten Thank you so much for your answer!
I tried with `{{}}`, as you told me, unfortunately it doesn't work either.
However, as you can see, I get newly created user ID.
![CleanShot 2023-01-23 at 16 09 00](https://user-images.githubusercontent.com/63867332/214074780-48ecf759-9b8d-46ce-ac6f-06f89219cd02.png)

**Comment by Corentin7301** [comment]
When I do this in Postman or Insomnia with GraphQL, it works well.
<img width="350" alt="CleanShot 2023-01-23 at 16 26 06@2x" src="https://user-images.githubusercontent.com/63867332/214078759-f3c7648a-d167-4b57-a836-fa413cdc3c8f.png">
